packages/yswaps/scripts/execute-fantom-pending-trades.ts

```diff
@@ -1,130 +1,105 @@
-    //   await tradeFactory.expire(pendingTrade._id);
-    //   continue;
-    // }
-
-    const { data: spookyData, minAmountOut: spookyMinAmountOut } = await uniswap.getBestPathEncoded({
-      tokenIn: pendingTrade._tokenIn,
-      tokenOut: pendingTrade._tokenOut,
-      amountIn: pendingTrade._amountIn,
-      uniswapV2Router: SPOOKYSWAP_ROUTER,
-      uniswapV2Factory: SPOOKYSWAP_FACTORY,
-      hopTokensToTest: [SPOOKY_TOKEN, WFTM, WETH],
-      slippage: SLIPPAGE_PERCENTAGE,
-    });
-
-    tradesSetup.push({
-      swapper: (await ethers.getContract('AsyncSpookyswap')).address,
-      swapperName: 'AsyncSpookyswap',
-      data: spookyData,
-      minAmountOut: spookyMinAmountOut,
-    });
-
-    console.log('spooky:', utils.formatUnits(spookyMinAmountOut!, decimalsOut), symbolOut);
-
-    const { data: spiritData, minAmountOut: spiritMinAmountOut } = await uniswap.getBestPathEncoded({
-      tokenIn: pendingTrade._tokenIn,
-      tokenOut: pendingTrade._tokenOut,
-      amountIn: pendingTrade._amountIn,
-      uniswapV2Router: SPIRITSWAP_ROUTER,
-      uniswapV2Factory: SPIRITSWAP_FACTORY,
-      hopTokensToTest: [SPIRIT_TOKEN, WFTM, WETH],
-      slippage: SLIPPAGE_PERCENTAGE,
-    });
-
-    tradesSetup.push({
-      swapper: (await ethers.getContract('AsyncSpiritswap')).address,
-      swapperName: 'AsyncSpiritswap',
-      data: spiritData,
-      minAmountOut: spiritMinAmountOut,
-    });
-
-    console.log('spirit:', utils.formatUnits(spiritMinAmountOut!, decimalsOut!), symbolOut);
-
-    const { data: zrxData, minAmountOut: zrxMinAmountOut } = await zrx.quote({
-      chainId: Number(chainId),
-      sellToken: pendingTrade._tokenIn,
-      buyToken: pendingTrade._tokenOut,
-      sellAmount: pendingTrade._amountIn,
-      slippagePercentage: SLIPPAGE_PERCENTAGE / 100,
-    });
-
-    tradesSetup.push({
-      swapper: (await ethers.getContract('ZRX')).address,
-      swapperName: 'ZRX',
-      data: zrxData,
-      minAmountOut: zrxMinAmountOut,
-    });
-
-    console.log('zrx:', utils.formatUnits(zrxMinAmountOut!, decimalsOut), symbolOut);
-
-    let bestSetup: TradeSetup = tradesSetup[0];
-
-    for (let i = 1; i < tradesSetup.length; i++) {
-      if (tradesSetup[i].minAmountOut!.gt(bestSetup.minAmountOut!)) {
-        bestSetup = tradesSetup[i];
-      }
-    }
+  // We set this so hardhat-deploys uses the correct deployment addresses.
+  process.env.HARDHAT_DEPLOY_FORK = 'fantom';
+  await evm.reset({
+    jsonRpcUrl: getNodeUrl('fantom'),
+  });
+
+  const ymech = new ethers.Wallet(await kms.decrypt(process.env.FANTOM_1_PRIVATE_KEY as string), ethers.provider);
+  await ethers.provider.send('hardhat_setBalance', [ymech.address, '0xffffffffffffffff']);
+  console.log('[Setup] Executing with address', ymech.address);
+
+  // We create a provider thats connected to a real network, hardhat provider will be connected to fork
+  fantomProvider = new ethers.providers.JsonRpcProvider(getNodeUrl('fantom'));
```

Comment:
on mainnet execute script this like is:
`mainnetProvider = new ethers.providers.JsonRpcProvider(getNodeUrl('mainnet'), 'mainnet');`

Not sure if needed here.

Reply:
nope. that would be for mainnet. this fantom sir. although is wrongly named mainnet, should be ethereum, since both chains have "mainnet and testnets" but, you get what I mean.

Reply:
Ahhhh, sorry. I see what you mean. The second argument. Yes, not needed. Weird ethers behavior.

Reply:
Lol sorry, my comment was so random. Yeah i meant the second argument with the network name.